### PR TITLE
Handle transaction memos

### PR DESCRIPTION
**Pull request by ThibautBremand**

### CODE REVIEW

File: packages/extension/cypress/e2e/send_token.cy.ts [hunk 1]
```diff
@@ -33,59 +33,19 @@ describe('Send Token', () => {
   });
 
   it('Send XRP', () => {
-    // Add receipient address
```
[the remark this answers is not in the document]
Comment (ThibautBremand) [reply packages/extension/cypress/e2e/send_token.cy.ts:36]: NP!

File: packages/extension/src/components/pages/SendPayment/ConfirmPayment/ConfirmPayment.tsx [hunk 2]
```diff
@@ -51,7 +54,8 @@ export const ConfirmPayment: FC<ConfirmPaymentProps> = ({
                 value: amount
               }
             : xrpToDrops(amount),
-        Destination: address
+        Destination: address,
+        Memos: buildMemos(memo)
```
[the remark this answers is not in the document]
Comment (ThibautBremand) [reply packages/extension/src/components/pages/SendPayment/ConfirmPayment/ConfirmPayment.tsx:56]: Please also take a look at [this commit](https://github.com/GemWallet/gemwallet-extension/pull/144/commits/eb0b600eeebf8b2d356bfef92da0d670731bf1b2)

File: packages/extension/src/components/pages/SendPayment/ConfirmPayment/ConfirmPayment.tsx [hunk 3]
```diff
@@ -6,6 +6,8 @@ import * as Sentry from '@sentry/react';
 import { useNavigate } from 'react-router-dom';
 import { xrpToDrops } from 'xrpl';
 
+import { buildMemos } from "@gemwallet/constants";
```
[the remark this answers is not in the document]
Comment (ThibautBremand) [reply packages/extension/src/components/pages/SendPayment/ConfirmPayment/ConfirmPayment.tsx]: Moved into the utils package

File: packages/constants/src/payload/payload.types.ts [hunk 4]
```diff
@@ -96,6 +98,21 @@ export interface NFTResponsePayload {
   nfts: AccountNFToken[] | null | undefined;
 }
 
+export const DEFAULT_MEMO_TYPE = 'Description';
+
+export const buildMemos = (memoData: string | undefined): { Memo: { MemoType: string; MemoData: string } }[] => {
+  if (memoData === undefined) return [];
+  return [
```
[the remark this answers is not in the document]
Comment (ThibautBremand) [reply packages/constants/src/payload/payload.types.ts]: Done, look [at this file](https://github.com/GemWallet/gemwallet-extension/pull/144/files#diff-19cfab74f3236dae08ca8a426a73d79252745aa9487fe4e4de5adc38083d711d)

File: packages/constants/src/payload/payload.types.ts [hunk 5]
```diff
@@ -96,6 +98,21 @@ export interface NFTResponsePayload {
   nfts: AccountNFToken[] | null | undefined;
 }
 
+export const DEFAULT_MEMO_TYPE = 'Description';
+
+export const buildMemos = (memoData: string | undefined): { Memo: { MemoType: string; MemoData: string } }[] => {
+  if (memoData === undefined) return [];
```
[the remark this answers is not in the document]
Comment (ThibautBremand) [reply packages/constants/src/payload/payload.types.ts]: Done, look [at this file](https://github.com/GemWallet/gemwallet-extension/pull/144/files#diff-19cfab74f3236dae08ca8a426a73d79252745aa9487fe4e4de5adc38083d711d)

File: packages/constants/src/payload/payload.types.ts [hunk 6]
```diff
@@ -96,6 +98,8 @@ export interface NFTResponsePayload {
   nfts: AccountNFToken[] | null | undefined;
 }
 
+export const DEFAULT_MEMO_TYPE = 'Description';
```
Comment (FlorianBouron) [inline packages/constants/src/payload/payload.types.ts:101]: This is a constant and not a type, I don't think it should live in this file.

File: packages/extension/src/utils/payment.ts [hunk 7]
```diff
@@ -0,0 +1,13 @@
+import { DEFAULT_MEMO_TYPE } from '@gemwallet/constants';
+
+export const buildMemos = (memoData: string | undefined): { Memo: { MemoType: string; MemoData: string } }[] | undefined => {
```
Comment (FlorianBouron) [inline packages/extension/src/utils/payment.ts:3]: Could you please unit test this file?

File: packages/extension/src/components/pages/SendPayment/ConfirmPayment/ConfirmPayment.tsx [hunk 8]
```diff
@@ -152,6 +155,20 @@ export const ConfirmPayment: FC<ConfirmPaymentProps> = ({
         <Typography variant="body1">Destination:</Typography>
         <Typography variant="body2">{address}</Typography>
       </Paper>
+      {memo && (
```
[the remark this answers is not in the document]
Comment (ThibautBremand) [reply packages/extension/src/components/pages/SendPayment/ConfirmPayment/ConfirmPayment.tsx]: Done

File: packages/extension/src/utils/payment.ts (hunk 7 above)
Comment (ThibautBremand) [reply packages/extension/src/utils/payment.ts:3]: Done

File: packages/constants/src/payload/payload.types.ts (hunk 6 above)
Comment (ThibautBremand) [reply packages/constants/src/payload/payload.types.ts]: Done [here](https://github.com/GemWallet/gemwallet-extension/pull/144/commits/1e35ec5f0af03a43e69142316981a9a5399715bc)

File: packages/constants/src/payload/payload.constant.ts [hunk 9]
```diff
@@ -0,0 +1 @@
+export const DEFAULT_MEMO_TYPE = 'Description';
```
Comment (FlorianBouron) [inline packages/constants/src/payload/payload.constant.ts:1]: This is used only in the "extension" package, let's move it within `/gemwallet-extension/packages/extension/src/constants`, could you please add an end of line?

Comment (ThibautBremand) [reply packages/constants/src/payload/payload.constant.ts]: Done [here](https://github.com/GemWallet/gemwallet-extension/pull/144/commits/0396971632122f29b0104e455a6a9c49ea59a336)

**Pull request merged**